src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -140,6 +144,22 @@ export class WorkspaceSavedObjectsClientWrapper {
     return false;
   };
 
+  private isDashboardAdmin(request: OpenSearchDashboardsRequest): boolean {
+    const config = this.config || ({} as WorkspacePluginConfigType);
+
+    // Before the user login OSD, calling the getPrincipalsFromRequest method will throw 'NOT_AUTHORIZED' exception.
+    try {
+      const { groups = [], users = [] } = this.permissionControl.getPrincipalsFromRequest(request);
```

Comment:
Wrap `try catch` on ` getPrincipalsFromRequest` is good enough I think.

Reply:
Yes, I had done it.